OZprivate/rawJS/OZTreeModule/src/controller/controller.js

```diff
@@ -124,13 +124,12 @@ class Controller {
       call_hook("before_draw");
       if ((tree_state.widthres != this.canvas.clientWidth)||(tree_state.heightres != this.canvas.clientHeight))
```

Comment:
Yeah good point I guess we could additionally check if `tree_state.widthres * window.devicePixelRatio == this.canvas.width`.
Not sure if it would then be necessary to add the listener since we'd be checking every frame anyway?

Reply:
Technically it is worth checking, the treeviewer will fall idle if nothing is happening, so we wouldn't notice a change in devicePixelRatio until something else wakes up the treeviewer. A window resize (e.g.) explicitly wakes the treeviewer up:

https://github.com/OneZoom/OZtree/blob/062a78ae58a77f2387c76ac8077982f50859f695/OZprivate/rawJS/OZTreeModule/src/interactor/interactor.js#L9-L21

But in reality this is overkill for what will be an incredibly subtle effect (Move window from a HDto 4k monitor, OneZoom is slightly blurry until you interact with it in some other way). I think what's in this pull is plenty.

What would be better is not checking canvas resolution every frame, and instead reacting to events, but that's a problem for another day.

Reply:
I did end up adding a listener for devicePixelRatio changes and not checking the canvas resolution every frame. 
Checking every frame was a bit worse than I thought: checking `canvas.clientWidth` forces certain element layout stuff to get pulled further forward in the browser rendering pipeline so it really isn't great for performance. It's not too bad on the main branch but it interacted badly with my other PR that updates the style.cursor since then the browser actually has to check whether it needs to update the width.